core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -327,27 +339,94 @@ private String getConfigurationFileName() {
         ClassLoader classLoader = TelemetryConfigurationFactory.class.getClassLoader();
         URL resource = classLoader.getResource(fileToParse);
 
-        // If not found as a resource, trying to load from the executing jar directory
         if (resource == null) {
```

Comment:
Your code would be more readable if you'll start with handling the case where `resource != null` as this will eliminate the need for an `else`:

    URL resource = classLoader.getResource(fileToParse); 
    if (resource != null) { 
        String configurationFile = resource.getFile(); 
        InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.INFO, "Found configuration file: '%s'", configurationFile); 
        return configurationFile; 
    } 

    InternalLogger.INSTANCE.logConfig(InternalLogger.LoggingLevel.INFO, "Did not find resource configuration file"); 
    // If not found as a resource, trying to load from the executing jar directory 
    String configurationFile = getConfFromSourceLocation(); 
    if (configurationFile != null) { 
        return configurationFile; 
    } 
 
    // If not found as a resource, trying to load from the class path 
    return getConfFromClassPath(classLoader);